docs/standard/simd.md

```diff
@@ -0,0 +1,103 @@
+
+## How to use SIMD?
+
+Before executing custom SIMD algorithms, it's possible to check if the host machine supports SIMD by using <xref:System.Numerics.Vector.IsHardwareAccelerated?displayProperty=nameWithType>, which returns a <xref:System.Boolean>. This doesn't guarantee that SIMD-acceleration is enabled for a specific type, but is an indicator that it's supported by some types.
+
+## Simple Vectors
+
+The most primitive SIMD-accelerated types in .NET are <xref:System.Numerics.Vector2>, <xref:System.Numerics.Vector3>, and <xref:System.Numerics.Vector4> types, which represent vectors with 2, 3, and 4 <xref:System.Single> values. The example below uses <xref:System.Numerics.Vector2> to add two vectors.
+
+```csharp
+var v1 = new Vector2(0.1f, 0.2f);
+var v2 = new Vector2(1.1f, 2.2f);
+var vResutl = v1 + v2;
+```
+
+It's also possible to use .NET vectors to calculate other mathematical properties of vectors such as `Dot product`, `Transform`, `Clamp` and so on.
+
+```csharp
+var v1 = new Vector2(0.1f, 0.2f);
+var v2 = new Vector2(1.1f, 2.2f);
+var vResutl1 = Vector2.Dot(v1, v2);
+var vResutl2 = Vector2.Distance(v1, v2);
+var vResutl3 = Vector2.Clamp(v1, Vector2.Zero, Vector2.One);
+```
+
+## Matrix
+
+<xref:System.Numerics.Matrix3x2>, which represents a 3x2 matrix, and <xref:System.Numerics.Matrix4x4>, which represents a 4x4 matrix. Can be used for matrix-related calculations. The example below demonstrates multiplication of a matrix to its correspondent transpose matrix using SIMD.
+
+```csharp
+var m1 = new Matrix4x4(
+            1.1f, 1.2f, 1.3f, 1.4f,
+            2.1f, 2.2f, 3.3f, 4.4f,
+            3.1f, 3.2f, 3.3f, 3.4f,
+            4.1f, 4.2f, 4.3f, 4.4f);
+
+var m2 = Matrix4x4.Transpose(m1);
+var mResult = Matrix4x4.Multiply(m1, m2);
+```
+
+## Vector\<T>
+
+The <xref:System.Numerics.Vector%601> gives the ability to use longer vectors. The count of a <xref:System.Numerics.Vector%601> instance is fixed, but its value <xref:System.Numerics.Vector%601.Count%2A?displayProperty=nameWithType> depends on the CPU of the machine running the code.
+
+The example below demonstrates adding long arrays elements using <xref:System.Numerics.Vector%601>.
+
+```csharp
+double[] SimdVectorProd(double[] left, double[] right)
+{
+    var offset = Vector<double>.Count;
+    double[] result = new double[left.Length];
+    int i = 0;
+    for (i = 0; i < left.Length; i += offset)
+    {
+        var v1 = new Vector<double>(left, i);
+        var v2 = new Vector<double>(right, i);
+        (v1 * v2).CopyTo(result, i);
+    }
+
+    //remaining items
+    for (; i < left.Length; ++i)
+    {
+        result[i] = left[i] * right[i];
+    }
+
+    return result;
+}
+```
+
+## Remarks
+
+SIMD is more likely to remove one bottleneck and expose the next, for example memory throughput. In general the performance benefit of using SIMD varies depending on the specific scenario, and in some cases it can even perform worse than simpler non-SIMD equivalent code.
```

Comment:
End with a note that users should benchmark and profile their code to identify hot spots and ensure that changes are meeting their performance goals/needs?

Can we link to https://github.com/dotnet/BenchmarkDotNet?